.github/workflows/terraform-observe_release.yaml

```diff
@@ -16,7 +16,20 @@ on:
         default: '0'
 
 jobs:
+  check-branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.get_branch.outputs.branch }}
+    steps:
+      - name: get_branch
+        id: get_branch
+        run: |
+          echo "branch=$(echo ${GITHUB_REF##*/})" >> $GITHUB_OUTPUT
```

Comment:
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

See `ref_name` and `ref_type`. Avoid using the environment variables when a suitable context variable exists.